core/src/main/java/dk/alexandra/fresco/lib/generic/CommitmentComputation.java

```diff
@@ -1,27 +1,27 @@
-package dk.alexandra.fresco.suite.spdz2k.protocols.computations;
+package dk.alexandra.fresco.lib.generic;
 
 import dk.alexandra.fresco.commitment.HashBasedCommitment;
 import dk.alexandra.fresco.framework.DRes;
 import dk.alexandra.fresco.framework.builder.Computation;
 import dk.alexandra.fresco.framework.builder.numeric.ProtocolBuilderNumeric;
 import dk.alexandra.fresco.framework.network.serializers.ByteSerializer;
 import dk.alexandra.fresco.framework.util.Drbg;
-import dk.alexandra.fresco.suite.spdz2k.protocols.natives.InsecureBroadcastProtocol;
+import dk.alexandra.fresco.framework.util.Pair;
 import java.util.ArrayList;
 import java.util.List;
 
 /**
- * Protocol for all parties to commit to a value and open it to the other parties.
+ * Protocol for all parties to commit to a value each and open it to the other parties.
  */
-public class Spdz2kCommitmentComputation implements
+public class CommitmentComputation implements
     Computation<List<byte[]>, ProtocolBuilderNumeric> {
 
   private final ByteSerializer<HashBasedCommitment> commitmentSerializer;
   private final byte[] value;
   private final int noOfParties;
```

Comment:
you can retrieve the `noOfParties` from the `builder` in `buildComputation`  so you don't need to pass it into the constructor, same for `CoinTossingComputation` (which only needs it to pass it into the constructor of `CommitmentComputation`)

```java
final int noOfParties = builder.getBasicNumericContext().getNoOfParties();
```